src/zenml/cli/artifact.py

```diff
@@ -58,123 +131,84 @@ def list_artifacts(**kwargs: Any) -> None:
-        only_metadata: If set, only delete metadata and not the actual artifact.
-        yes: If set, don't ask for confirmation.
+        name_id_or_prefix: Either the ID of the artifact version or the name of
+            the artifact.
+        version: The version of the artifact to get. Only used if
+            `name_id_or_prefix` is the name of the artifact. If not specified,
+            the latest version is returned.
+        tag: Tags to add to the artifact version.
+        remove_tag: Tags to remove from the artifact version.
     """
-    is_uuid = is_valid_uuid(artifact_name_or_id)
-
-    if not yes:
-        if is_uuid:
-            confirmation = cli_utils.confirmation(
-                f"Are you sure you want to delete artifact "
-                f"'{artifact_name_or_id}'?"
-            )
-        elif version:
-            confirmation = cli_utils.confirmation(
-                f"Are you sure you want to delete version '{version}' of "
-                f"artifact '{artifact_name_or_id}'?"
-            )
-        else:
-            confirmation = cli_utils.confirmation(
-                f"Are you sure you want to delete all versions of artifact "
-                f"'{artifact_name_or_id}'?"
-            )
-        if not confirmation:
-            cli_utils.declare("Artifact deletion canceled.")
-            return
-
     try:
-        versions: List[Optional[str]] = [version]
-        if not is_uuid and not version:
-            artifact_models = depaginate(
-                partial(Client().list_artifacts, name=artifact_name_or_id)
-            )
-            versions = [
-                str(artifact_model.version)
-                for artifact_model in artifact_models
-            ]
-        for version_ in versions:
-            Client().delete_artifact(
-                name_id_or_prefix=artifact_name_or_id,
-                version=version_,
-                delete_metadata=not only_artifact,
-                delete_from_artifact_store=not only_metadata,
-            )
+        artifact_version = Client().update_artifact_version(
+            name_id_or_prefix=name_id_or_prefix,
+            version=version,
+            add_tags=tag,
+            remove_tags=remove_tag,
+        )
     except (KeyError, ValueError) as e:
         cli_utils.error(str(e))
     else:
-        if is_uuid:
-            cli_utils.declare(f"Artifact '{artifact_name_or_id}' deleted.")
-        elif version:
-            cli_utils.declare(
-                f"Version '{version}' of artifact '{artifact_name_or_id}' "
-                f"deleted."
-            )
-        else:
-            cli_utils.declare(
-                f"All versions of artifact '{artifact_name_or_id}' deleted."
-            )
+        cli_utils.declare(f"Artifact version '{artifact_version.id}' updated.")
 
 
-@artifact.command("prune", help="Delete all unused artifacts.")
+@artifact.command(
+    "prune", help="Delete all unused artifacts and artifact versions."
```

Comment:
Here and/or below, might be worth below specifying that 'unused' means no longer associated with a pipeline run.

Also, are there any lineage concerns that people need to be aware of when using this command? i.e. can running this make it impossible to reconstruct previous pipelines / runs etc?

Reply:
Makes sense. No, the command only deletes artifacts that can't break the lineage, that's exactly the point of this command and why I removed the option to delete artifacts manually

Reply:
Yeah the main thing I was noting was that 'unused' isn't that clear / descriptive.